group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetrics.java

```diff
@@ -281,100 +297,93 @@ public void onUpdateLastCommittedOffset(TopicPartition tp, long offset) {
         }
     }
 
-    public static MetricName getMetricName(String type, String name) {
+    public static com.yammer.metrics.core.MetricName getMetricName(String type, String name) {
         return getMetricName("kafka.coordinator.group", type, name);
     }
 
     private void registerGauges() {
-        registry.newGauge(NUM_OFFSETS, new Gauge<Long>() {
+        registry.newGauge(NUM_OFFSETS, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
                 return numOffsets();
             }
```

Comment:
nit: Could we also use a lambda for those in order to make the code consistent with the other metrics?

Reply:
the format is different between yammer metrics and kafka metrics. numOffsets follow the yammer format

Reply:
just to be clear, i was wondering whether the following would work:

```
registry.newGauge(NUM_OFFSETS, () -> numOffsets());
```

Does it?

Reply:
Right, that does not work. I believe it's because yammer Gauge is not a functional interface